doc/external_resources.md

```diff
@@ -0,0 +1,20 @@
+# External Resources
+
+Below is a curated collection of external PyGMT resources.
+
+*To add your contribution to this collection, submit a recommendation via a
+[pull request](https://github.com/GenericMappingTools/pygmt/blob/master/doc/external_resources.md).*
```

Comment:
I think adding to the documentation editing instructions would be helpful here and am open to your preferred wording.
```suggestion
*To add your contribution to this collection, follow [these instructions](https://github.com/GenericMappingTools/pygmt/blob/master/CONTRIBUTING.md#editing-the-documentation)
to submit a pull request with your recommended addition to the
[External Resources file](https://github.com/GenericMappingTools/pygmt/blob/master/doc/external_resources.md).*
```